openhands-sdk/openhands/sdk/llm/utils/model_features.py

```diff
@@ -15,6 +15,28 @@ def model_matches(model: str, patterns: list[str]) -> bool:
     return False
 
 
+def apply_ordered_model_rules(model: str, rules: list[str]) -> bool:
+    """Apply ordered include/exclude model rules to determine final support.
+
+    Rules semantics:
+    - Each entry is a substring token. '!' prefix marks an exclude rule.
+    - Case-insensitive substring matching against the raw model string.
+    - Evaluated in order; the last matching rule wins.
+    - If no rule matches, returns False.
+    """
+    raw = (model or "").strip().lower()
+    decided: bool | None = None
+    for rule in rules:
+        token = rule.strip().lower()
+        if not token:
+            continue
+        is_exclude = token.startswith("!")
+        core = token[1:] if is_exclude else token
+        if core and core in raw:
+            decided = not is_exclude
+    return bool(decided)
```

Comment:
@OpenHands do not change any code, do you know if there's some python library that already provides these support?